### PR TITLE
austin.intermediate.summer2.mccalmont.dave

### PR DESCRIPTION
**Pull request by DMcCal79**

### CODE REVIEW

**Comment by paulrenenichols** [comment]
Please rename your pul request. Here's our convention for this class:

```
austin.intermediate.summer2.lastname.firstname`
```

**Comment by paulrenenichols** [comment]
Not seeing your Pit Latin work here.

The goal for the first app is to replace your apps/01PigLatin.js file with my code from [here](https://gist.github.com/paulrenenichols/099c38009a569d603a098060f6cc3018), and then fill in the concept checks.

File: homework/02LessonTwo.js [hunk 1]
```diff
@@ -26,6 +26,7 @@ var alwaysTrue = function () {
 // argument1 is equal to argument2
 
 function equals(argument1, argument2) {
+  return argument1 === argument2;
```
Comment (paulrenenichols) [inline homework/02LessonTwo.js:29]: nice!

File: apps/01PigLatin.js [hunk 2]
```diff
@@ -7,11 +7,80 @@ prompt.start();
...
 
-    // Your code here
+  // Concept Check
+  //
+  // Please explain why we are using String.prototype.toLowerCase() here.
+  // For more info:
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/toLowerCase
+  //
+  // Your Answer:
+  // This will convert the string to all lowercase letters.  This is important because the conditionals are case-sensitive.  For example, if you were trying to find the index of 'a' in the word 'Apple', then the result would be -1.  This is becuase the conditional is looking for a lowercase 'a' and the one in 'Apple' is capitalized.
```
Comment (paulrenenichols) [inline apps/01PigLatin.js:17]: Good

File: apps/01PigLatin.js [hunk 3]
```diff
@@ -7,11 +7,80 @@ prompt.start();
...
+  word  = word.toLowerCase();
 
+  // Concept Check
+  //
+  // What is the significance of setting vowelIndex to -1?
+  // This might help:
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/lastIndexOf
+  //
+  // Your Answer:
+  // Setting the vowelIndex to -1 will allow us to know if there weren't any vowels in the string.  If we end up with a value of -1 after all of the conditionals have been run, then this tells us that there weren't any vowels found in the string.
```
Comment (paulrenenichols) [inline apps/01PigLatin.js:27]: Good answer. also **indexOf()** returns -1 if it can't find anything.

File: homework/02LessonTwo.js [hunk 4]
```diff
@@ -54,22 +59,23 @@ function lessThanOrEqualTo(parameter1, parameter2) {
...
 // Your Answer Goes Here:
-//
+//A function takes inputs from a user, does something with the inputs, and produces a desired output usually indicated after the "return" command.  In Javascript, a function can be defined in a couple of ways.  The first way is: function functionName (parameter1, parameter2) or this way: var anotherFunction = function (parameter3, parameter4)
 //
 // What is a return value?
 //
 // Your Answer Goes Here:
-//
+//The return value is what you want is want in return.  In other words, it's what you want happend to the parameters indicated in function declaration.  For example: return parameter1 + parameter2;
 //
```
Comment (maddiesolz) [inline homework/02LessonTwo.js:68]: Great work! All your tests are passing :)

File: homework/03LessonThree.js [hunk 5]
```diff
@@ -110,10 +110,10 @@ var threeByThreeArray = [
 
 // Problem 11:
 // Replace the center element of threeByThreeArray, which is 5, with something else.
-
+threeByThreeArray[1][1] = 20;
 // Problem 12:
```
Comment (maddiesolz) [inline homework/03LessonThree.js:114]: Very good! This is the answer we are looking for: Indexing the 2d array.

File: apps/03TicTacToe.js [hunk 6]
```diff
@@ -22,24 +22,64 @@ function printBoard() {
...
-    // Your code here
-}
+    board[row][column] = playerTurn;
+    checkForWin();
+    playerTurn = (playerTurn === 'X') ? 'O' : 'X';
+};
+
 
 function getPrompt() {
     printBoard();
```
Comment (maddiesolz) [inline apps/03TicTacToe.js:85]: Great job on the app! It is working as expected :) When you have some spare time, try to do some input scrubbing to account for inputs like "01" or "0,1". Keep it up!

Comment (DMcCal79) [inline apps/03TicTacToe.js:85]: Thanks! I'll give that a shot.

File: homework/04LessonFour.js [hunk 7]
```diff
@@ -87,7 +95,7 @@ var nestedObjects = {
 // What is an associative array? What is the difference between an array and an associative array?
 //
 // Your Answer Goes Here:
-//
+// An object (associate array) is a collection of data that is stored in properties. A property is made up of a key and a value.  For example, a key of city could have the value of 'Austin'.  The main difference between an array an an associate array (object) is how the data is stored.  An array uses a 0-index while an object uses properties made up of keys & values.
```
Comment (maddiesolz) [inline homework/04LessonFour.js:98]: Great work Dave!

File: apps/04TowersOfHanoi.js [hunk 8]
```diff
@@ -10,30 +10,51 @@ var stacks = {
...
+    var endStackArray = stacks[endStack];
+
+    if (startStackArray.length === 0) {
+      return false;
+    }
+    else if (endStackArray.length === 0) {
+      return true;
+    }
+    else {
+      return startStackArray[startStackArray.length -1] < [endStackArray.length - 1]
```
Comment (maddiesolz) [inline apps/04TowersOfHanoi.js:38]: Very good!

Comment (maddiesolz) [inline apps/04TowersOfHanoi.js:38]: but missing a ";"

File: apps/04TowersOfHanoi.js [hunk 9]
```diff
@@ -10,30 +10,51 @@ var stacks = {
...
+    return false;
+  }
 }
 
 function towersOfHanoi(startStack, endStack) {
     // Your code here
-
+    if (isLegal(startStack, endStack)) {
+      movePiece(startStack, endStack);
+      checkForWin(startStack, endStack);
```
Comment (maddiesolz) [inline apps/04TowersOfHanoi.js:56]: Awesome!

File: homework/05LessonFive.js [hunk 10]
```diff
@@ -63,15 +63,15 @@ for (forLoopCounter = 0; forLoopCounter > -10; forLoopCounter--) {
...
+var forLoopArrayResult = [0,-1,-2,-3,-4,-5,-6,-7,-8,-9];
 
 // ****
 // Concept Checkpoint
 //
 // Write your answer in comments
 //
 // What are loops? Why do we use them?
-//
+// A loop will first check a condition in the ().  If the condition returns true, then the code block in the {} will run.  After the code block has run, then the condition in the () will be checked again and if it still returns true, the code block will run again.  This process repeats until the condition in the () returns false. Loops help save time by reducing the # of lines code that need to be written. A process that would normally several lines of code could be written in just a few lines with a loop.
```
Comment (maddiesolz) [inline homework/05LessonFive.js:74]: Sounds great!

File: apps/05Mastermind.js [hunk 11]
```diff
@@ -26,13 +26,43 @@ function getRandomInt(min, max) {
...
+    var correctLetterLocations = 0;
+    for (var i = 0; i < solutionArray.length; i++) {
+      if (solutionArray[i] === guessArray[i]) {
+        correctLetterLocations++;
+        solutionArray[i] = null;
+        }
+    }
+    var correctLetters = 0;
+    for (var i = 0; i < solutionArray.length; i++) {
+      var targetIndex = guessArray.indexOf(solutionArray[i]);
```
Comment (maddiesolz) [inline apps/05Mastermind.js:42]: Great work!! Would this work if I did solutionArray.indexOf(guessArray[i])?

File: homework/06LessonSix.js [hunk 12]
```diff
@@ -36,7 +40,7 @@ var anotherPoint;
 //
 // What is the difference between point (an object literal, or associative array)
 // and anotherPoint (a point object constructed from the Point class)?
-//
+// The variable anotherPoint is defined by using an object constructor instead of object literal. An object constructor is a function that will take the entered parameters and assign them to the correct properties.  In other words, an object constructor is like a template that tells the computer which parameters go with which property.  Using an object constructor saves time when having to create multitple objects with similar properties.
```
Comment (maddiesolz) [inline homework/06LessonSix.js:43]: Very good!

File: homework/06LessonSix.js [hunk 13]
```diff
@@ -76,27 +80,30 @@ function ConferenceRoom() {
...
+conferenceRoom.enter(jen);
 
 // ****
 // Concept Checkpoint
 //
 // Write your answer in comments
 //
 // What is a method?
-//
+// A method is a function that is part of an object.  They are tasks that are associated with an object and are listed as the value of a property. For example, you can have an object for a baseball player that had the properties and values for this.atBats and this.hits. Then you can have a method for calculating batting average such as: this.batAvg = function() {return this.hits / this.atBats;} .
```
[the remark this answers is not in the document]
Comment (maddiesolz) [reply homework/06LessonSix.js:106]: Yes!!!

File: homework/07LessonSeven/script.js [hunk 14]
```diff
@@ -69,20 +69,22 @@ $(document).on('ready', function() {
   // Problem 9:
   // select the element with id="insert-text"
   // then set the text of this element to "Text Has Been Inserted"
+  $('#insert-text').text('Text Has Been Inserted');
```
Comment (maddiesolz) [inline homework/07LessonSeven/script.js:72]: Good work on this homework and app6!! Hows OOP? Always read the tests to see why something that definitely should be passing isnt!

**Pull request closed without merging**